tests/lazy_test.go

```diff
@@ -30,20 +29,20 @@ func lazyCanLazilyLoadEntity(t *testing.T, driver *RavenTestDriver) {
 	{
 		session := openSessionMust(t, store)
 		query := session.Advanced().Lazily()
-		lazyOrder := query.Load(reflect.TypeOf(&Company{}), "companies/1", nil)
+		var order *Company
+		lazyOrder := query.Load(&order, "companies/1", nil)
 
 		assert.False(t, lazyOrder.IsValueCreated())
-		orderI, err := lazyOrder.GetValue()
+		err = lazyOrder.GetValue()
```

Comment:
`GetValue()` returns just an error? How do I get the actual value? Do I have to remember it?

Reply:
See the above comment for context.

Yes, to make this a strongly typed API, the caller has to pass a pointer to the value that will be set as a result of Lazy operation.

Since Go doesn't have generics, I can't return strongly typed value, only `interface{}` (the equivalent of C#'s `object`) so there's little point of returning it because in most cases the caller already has a  strongly typed reference to it.

There's one case in the code where that's not true so the value can be obtained via `Lazy.Value`

Reply:
I like it